src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -959,7 +960,19 @@ private static void OnAdditionalDocumentUpdatedOnDisk(object sender, EventArgs e
             string filename,
             SourceCodeKind sourceCodeKind,
             Func<IVisualStudioHostDocument, bool> getIsCurrentContext,
-            Func<uint, IReadOnlyList<string>> getFolderNames)
+            Func<uint, IReadOnlyList<string>> getFolderNames,
+            IDocumentServiceFactory documentServiceFactory)
+        {
+            AddFile(filename, sourceTextContainer: null, sourceCodeKind, getIsCurrentContext, getFolderNames, documentServiceFactory);
+        }
+
+        protected void AddFile(
+            string filename,
+            SourceTextContainer sourceTextContainer,
+            SourceCodeKind sourceCodeKind,
+            Func<IVisualStudioHostDocument, bool> getIsCurrentContext,
+            Func<uint, IReadOnlyList<string>> getFolderNames,
+            IDocumentServiceFactory documentServiceFactory)
```

Comment:
This looks like a bug. The `IDocumentServiceFactory` isn't passed through to the document provider.

Reply:
Same with the `SourceTextContainer` they are ignored.